experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,256 @@
+  l=size(charTirr)[1]
+  res=AbsSimpleNumFieldElem[]
+  a=gen(K)
+  for j in 1:l
+    for i in 1:l
+      if r > 2
+        coeff=CyclotomicNumbers.coefficients(charTirr[i,j])
+        temp=K(0)
+        for k in 1:length(coeff)
+          temp+=coeff[k]*a^(k-1)
+        end
+        push!(res,temp)
+      else
+        push!(res,K(charTirr[i,j]))
+      end
+    end
+  end
+  return reshape(res,l,l)
+end
+
+#Reoders the CharTable from Chevie
+function reorder(charTirr::Matrix{Cyc{Int64}}, r::Integer, n::Integer, Modules::Vector{Multipartition{Int64}}, K::AbsSimpleNumField)
+  mps=collect(multipartitions(n,r))
+  new_ord=Int64[]
+    for lbb in mps
+      i=findfirst(x-> x==lbb,Modules)
+      push!(new_ord,i)
+    end
+  return chev_to_osc(charTirr,r,K)[new_ord,:]
+end
+
+#Tools from representation theory
+
+# Computes the fake degree of a multipartition cf. Ste89 Thm 5.3 and Prop. 3.3.2 Haiman cdm
+function fake_deg(lbb::Multipartition, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem}}, var::AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem})
+  r=size(lbb)[1]
+  n=sum(lbb)[1]
+  res=Q(1)
+  for p in 1:n
+    res=res*(1-var^(r*p))
+  end
+  for k in 1:r
+    for i in 1:length(lbb[k])
+      for j in 1:lbb[k][i]
+        hookfactor=(Q(1)-var^(r*(1+lbb[k][i]+length(findall(row->row >= j,lbb[k]))-j-i)))
+        res=res//hookfactor
+      end
+    end
+  end
+  res=res*var^btot(lbb)
+  return res
+end
+
+# computes C_Delta defined in PhD relation (1.45)
+function C_Delta(r::Integer, n::Integer, K::AbsSimpleNumField, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem}}, var::AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem})
+  G=complex_reflection_group(r,1,n)
+  Modules=charinfo(G).charparams
+  charTable=CharTable(G)
+  charTirr=charTable.irr
+  Modules=[multipartition(map(x->partition(x),lbb[1])) for lbb in Modules]
+  l=length(Modules)
+  charTirr=reorder(charTirr,r,n,Modules,K)
+  mult=collect(multipartitions(n,r))
+  rows=zero_matrix(Q,l,0)
+  for i in 1:l
+    col=zero_matrix(Q,l,1)
+    for j in 1:l
+      v=matrix(K,l,1,map(k->charTirr[i,k]*charTirr[j,k],1:l))
+      x=solve(matrix(K,transpose(charTirr)),v,side=:right)
+      col=col+fake_deg(mult[j],Q,var)*x
+    end
+    rows=hcat(rows,col)
+  end
+  return rows
+end
+
+# computes the character of the simple representations of the Cherednik algebra.
+function C_L(r::Integer, n::Integer, K::AbsSimpleNumField, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem}}, var::AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem})
+  mps=collect(multipartitions(n,r))
+  l=length(mps)
+  C_D=C_Delta(r,n,K,Q,var)
+  diag=[]
+  for i in 1:l
+    for j in 1:l
+      if i==j
+        push!(diag,var^btot(mps[i])//fake_deg(mps[i],Q,var))
+      else
+        push!(diag,0)
+      end
+    end
+  end
+  diag=matrix(Q,l,l,diag)
```

Comment:
Instead of filling in all those zeros, you can use `diagonal_matrix`. And as Lars mentioned, no need to collect. And instead of iterating over indices, we may just as well iterate directly over `mps`. This all makes the code much more concise:
```suggestion
  C_D = C_Delta(r,n,K,Q,var)
  mps = multipartitions(n,r)
  d = [var^btot(mp)//fake_deg(mp,Q,var)) for mp in mps]
  diag = diagonal_matrix(Q,d)
```
This change also should improve type stability: in your original code, `diag` first was a `Vector{Any}` (so no type information about the elements), then it gets changed to a matrix -- unfortunately Julia then may have trouble correctly inferring the return value.

Reply:
Thanks a lot for this improvement.